### PR TITLE
Fix miscategorized scenes

### PR DESCRIPTION
**Pull request by cadmic**
* Move `ganon_tou` (Ouside Ganon's Castle) from "dungeons" to "overworld", like `spot15` (Hyrule Castle)
* Move `entra` (Child Market Entrance Day) from "overworld" to "misc", like `entra_n` (Child Market Entrance Night) and `enrui` (Adult Market Entrance Ruins)
* Move `souko` (Lon Lon Ranch House and Tower) from "overworld" to "indoors"

### CODE REVIEW

File: src/code/z_scene_table.c [hunk 1]
```diff
@@ -1,6 +1,8 @@
 #include "global.h"
 #include "quake.h"
 
+#include "assets/scenes/indoors/souko/souko_scene.h"
```
Comment (mzxrules) [inline src/code/z_scene_table.c:4]: not quite related to your changes but the `#include "assets/scenes/indoors/miharigoya/miharigoya_scene.h"` seems oddly positioned in this file

Comment (cadmic) [reply src/code/z_scene_table.c:4]: hm what would be a better positioning?

Comment (fig02) [reply src/code/z_scene_table.c:4]: the list seems to attempt to sort by category, but miharigoya is indoors, not dungeon. you can move it up to the indoors section you just created

Comment (cadmic) [reply src/code/z_scene_table.c:5]: oh I see, done

**Review by mzxrules — APPROVED** [review]

**Pull request merged**